src/utils.py

```diff
@@ -311,23 +288,32 @@ def __init__(self, name: str,
 
     def _initiable(self, state: State, memory: Dict, objects: Sequence[Object],
                    params: Array) -> bool:
-        # Reset the current child to the first one.
+        # Initialize the current child to the first one.
         memory["current_child_index"] = 0
+        # Create memory dicts for each child to avoid key collisions. One
+        # example of a failure that arises without this is when using
+        # action_sequence_to_parameterized_option, each of the singleton
+        # options would be referencing the same start_state in memory.
+        memory["child_memory"] = [{} for _ in self._children]
```

Comment:
good catch. there's never going to be anything else in `memory` that we need to copy over into the child memories, right?